ta_services/secure_key_services/include/sks_ta.h

```diff
@@ -0,0 +1,872 @@
+	uint32_t ulMaxSessionCount;
+	uint32_t ulSessionCount;
+	uint32_t ulMaxRwSessionCount;
+	uint32_t ulRwSessionCount;
+	uint32_t ulMaxPinLen;
+	uint32_t ulMinPinLen;
+	uint32_t ulTotalPublicMemory;
+	uint32_t ulFreePublicMemory;
+	uint32_t ulTotalPrivateMemory;
+	uint32_t ulFreePrivateMemory;
+	uint8_t hardwareVersion[2];
+	uint8_t firmwareVersion[2];
+	uint8_t utcTime[16];
+};
+
+/* Flags (reflect the cryptoki flags) */
+#define SKS_TOKEN_HAS_RNG		BIT(0UL)	/* CKF_RNG */
+#define SKS_TOKEN_IS_READ_ONLY		BIT(1UL)	/* CKF_WRITE_PRO... */
+#define SKS_TOKEN_REQUIRE_LOGIN		BIT(2UL)	/* CKF_LOGIN_REQ... */
+#define SKS_TOKEN_HAS_USER_PIN		BIT(3UL)	/* CKF_USER_PIN_... */
+#define SKS_TOKEN_FULLY_RESTORABLE	BIT(4UL)	/* CKF_RESTORE_K... */
+#define SKS_TOKEN_HAS_CLOCK		BIT(5UL)	/* CKF_CLOCK_ON_... */
+#define SKS_TOKEN_ALT_AUTHENT		BIT(6UL)	/* CKF_PROTECTED... */
+#define SKS_TOKEN_CAN_DUAL_PROC		BIT(7UL)	/* CKF_DUAL_CRYP... */
+#define SKS_TOKEN_INITED		BIT(8UL)	/* CKF_TOKEN_INI... */
+#define SKS_TOKEN_USR_PIN_FAILURE	BIT(9UL)	/* CKF_USER_PIN_... */
+#define SKS_TOKEN_USR_PIN_LAST		BIT(10UL)	/* CKF_USER_PIN_... */
+#define SKS_TOKEN_USR_PIN_LOCKED	BIT(11UL)	/* CKF_USER_PIN_... */
+#define SKS_TOKEN_USR_PIN_TO_CHANGE	BIT(12UL)	/* CKF_USER_PIN_... */
+#define SKS_TOKEN_SO_PIN_FAILURE	BIT(13UL)	/* CKF_SO_PIN_CO... */
+#define SKS_TOKEN_SO_PIN_LAST		BIT(14UL)	/* CKF_SO_PIN_FI... */
+#define SKS_TOKEN_SO_PIN_LOCKED		BIT(15UL)	/* CKF_SO_PIN_LO... */
+#define SKS_TOKEN_SO_PIN_TO_CHANGE	BIT(16UL)	/* CKF_SO_PIN_TO... */
+#define SKS_TOKEN_BAD_STATE		BIT(17UL)	/* CKF_ERROR_STATE */
+
+/*
+ * SKS_CMD_CK_MECHANISM_IDS - Get list of the supported mechanisms
+ *
+ * param#0: in(*)-memref : [uint32_t slot_id]
+ * param#1: none
+ * param#2: out-memref : [struct sks_ck_mecha_id mecha_ids[N]]
+ * param#3: none
+ *
+ * This commands relates to the PKCS#11 API function C_GetMechanismList.
+ */
+#define SKS_CMD_CK_MECHANISM_IDS	0x00000004
+
+struct sks_ck_mecha_id {
+	uint32_t mecha_id;
+};
+
+/*
+ * SKS_CMD_CK_MECHANISM_INFO - Get information on a specific mechanism
+ *
+ * param#0: in(*)-memref : [uint32_t slot_id]
+ *			   [struct sks_ck_mecha_id mecha_id]
+ * param#1: none
+ * param#2: out-memref : [struct sks_ck_mecha_info info]
+ * param#3: none
+ *
+ * This commands relates to the PKCS#11 API function C_GetMechanismInfo.
+ */
+#define SKS_CMD_CK_MECHANISM_INFO	0x00000005
+
+struct sks_ck_mecha_info {
+	uint32_t min_key_size;
+	uint32_t max_key_size;
+	uint32_t flags;
+};
+
+/* flags */
+#define SKS_PROC_HW			BIT(0UL)
+#define SKS_PROC_ENCRYPT		BIT(1UL)
+#define SKS_PROC_DECRYPT		BIT(2UL)
+#define SKS_PROC_DIGEST			BIT(3UL)
+#define SKS_PROC_SIGN			BIT(4UL)
+#define SKS_PROC_SIGN_RECOVER		BIT(5UL)
+#define SKS_PROC_VERIFY			BIT(6UL)
+#define SKS_PROC_VERFIY_RECOVER		BIT(7UL)
+#define SKS_PROC_GENERATE		BIT(8UL)
+#define SKS_PROC_GENERATE_PAIR		BIT(9UL)
+#define SKS_PROC_WRAP			BIT(10UL)
+#define SKS_PROC_UNWRAP			BIT(11UL)
+#define SKS_PROC_DERIVE			BIT(12UL)
```

Comment:
Same as above, could this be more consistent (mecha vs. mechanism vs. proc)?
```C
struct sks_mechanism_info {
	...
};

/*
 * Values for sks_mechanism_info::flags.
 * SKS_MECHANISM_<x> corresponds to CKF_<x>
 */
#define SKS_MECHANISM_HW        BIT(0)
#define SKS_MECHANISM_ENCRYPT   BIT(1)
...
```

Reply:
ok.